BackEnd/profiles/serializers.py

```diff
@@ -378,3 +381,31 @@ def get_user_profile_name(self, obj) -> str:
 
     def get_company_name(self, obj) -> str:
         return obj.company_name
+
+
+class ProfileModerationSerializer(serializers.Serializer):
+    uid = serializers.CharField()
+    action = serializers.CharField()
+    timestamp = serializers.IntegerField()
+
+    def validate_uid(self, value):
+        try:
+            Profile.objects.get(pk=decode_uid(value))
+        except Profile.DoesNotExist:
+            raise serializers.ValidationError("Profile does not exist")
+        return value
+
+    def validate_action(self, value):
+        if value not in ["approve", "reject"]:
```

Comment:
Please don't use such hardcode string values.
Use or Enum or StrEnum, or global variable to contain such values

Reply:
or serializers.EnumChoiceField() and global variable

Reply:
updated with Enum and ChoiceField